### ISSUE TITLE
Possible incorrect CID generated when preserving file attributes

### ISSUE THREAD

**Issue by kstuart** [ipfs/js-ipfs#4093]
js-ipfs version: 0.14.3
interface-ipfs-core version: ^0.154.3
ipfs-http-client version: 1.0.0
Repo version: 12
System version: x64/linux
Node.js version: v16.4.2

### Steps to reproduce:

**touch -d@1652299617 /tmp/empty_file**

**jsipfs add --preserve-mode --preserve-mtime /tmp/empty_file**
added QmcrCRhLSqsroGBX2PyBtdCrKWBDxjgPpATLg3KxMqywna empty_file

**jsipfs files stat /ipfs/QmcrCRhLSqsroGBX2PyBtdCrKWBDxjgPpATLg3KxMqywna**
QmcrCRhLSqsroGBX2PyBtdCrKWBDxjgPpATLg3KxMqywna
Size: 6
CumulativeSize: 22
ChildBlocks: 0
Type: file
Mode: -rw-r--r--
Mtime: May 11, 2022, 08:06:57 PM UTC

#### go-ipfs
**ipfs add --preserve-mode --preserve-mtime /tmp/empty_file**
added QmX5na5diwD9h9LzYdfkhxEaYAgCVkAgTVNUiFWkkxivKS empty_file

**ipfs files stat /ipfs/QmX5na5diwD9h9LzYdfkhxEaYAgCVkAgTVNUiFWkkxivKS**
QmX5na5diwD9h9LzYdfkhxEaYAgCVkAgTVNUiFWkkxivKS
Size: 0
CumulativeSize: 17
ChildBlocks: 0
Type: file
Mode: -rw-r--r--
Mtime: 11 May 2022, 21:06:57 BST

### Expected:

Assuming that the _Size_ reported in `jsipfs files stat` is the file data size, expected this value to be 0 for an empty file, and the CID to be the same as the one generated by go-ipfs. It's possible go-ipfs is generating the incorrect CID but the discrepancy in _Size_ value would indicate probably not.

**Comment by BigLep** [comment]
2022-06-10 conversation: there are two things to look into:
1. Why js-ipfs is returning a non-zero size
2. Why the CIDs are different

Maintainers agree this needs to get looked into.  There isn't an estimate yet on when this will get picked up though given other priorities.  Is this blocking for you @kstuart on your work?

**Comment by kstuart** [comment]
@BigLep no, this doesn't block me. After further investigation I've updated the issue, note that preserving mode or mtime will add erroneous file data to the node, essentially a corruption of the original file.

**Comment by whizzzkid** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can follow the migration plan here https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).

Please feel to reopen with any comments before 2023-06-05. We will do a final pass on reopened issues afterward (see https://github.com/ipfs/js-ipfs/issues/4336).

This issue might be resolved in Helia, if not, please let us know!
